pandas/core/generic.py

```diff
@@ -8988,26 +8992,38 @@ def _align_series(
                     other.index, how=join, level=level, return_indexers=True
                 )
 
-            left = self._reindex_indexer(join_index, lidx, copy)
+            if is_series:
+                left = self._reindex_indexer(join_index, lidx, copy)
+            elif lidx is None:
+                left = self.copy() if copy else self
+            else:
+                data = algos.take_nd(
+                    self.values,
```

Comment:
self.values can be expensive (and lossy) if the dataframe is not a single block.